pkg/apis/kops/parse.go

```diff
@@ -22,6 +22,27 @@ import (
 	"strings"
 )
 
+// ParseInstanceGroupRole converts a string to an InstanceGroupRole
+func ParseInstanceGroupRole(input string, lenient bool) (InstanceGroupRole, bool) {
+	findRole := strings.ToLower(input)
+	if lenient {
+		// Accept pluralized "bastions" for "bastion"
+		findRole = strings.TrimSuffix(findRole, "s")
+	}
+
+	for _, role := range AllInstanceGroupRoles {
+		s := string(role)
+		s = strings.ToLower(s)
+		if lenient {
+			s = strings.TrimSuffix(s, "s")
+		}
+		if s == findRole {
+			return role, true
+		}
+	}
+	return "", false
+}
```

Comment:
This is a different pattern I think.  It could return (*InstanceGroupRole), but it will never (currently) return an error.  But then we probably want to return (*InstanceGroupRole, error) in case it can return an error in future.  And then we start to realize why exceptions are a good idea ;-)